pkg/apis/hive/v1alpha1/validating-webhooks/syncset_validating_admission_hook.go

```diff
@@ -224,3 +245,43 @@ func checkValidPatchTypes(patches []hivev1.SyncObjectPatch) (string, bool) {
 	}
 	return "", true
 }
+
+func checkSecretReferences(secrets []hivev1.SecretReference) field.ErrorList {
+	allErrs := field.ErrorList{}
+	for i, secret := range secrets {
+		sourcePath := field.NewPath("spec").Child("secretReferences").Index(i).Child("source")
```

Comment:
Create a function to do the checks for the `ObjectReference` rather than duplicating it for source and target.

```
for i, secret := range secrets {
  validateSecretRef(secret.Source, fldPath.Index(i).Child("source")
  validateSecretRef(secret.Target, fldPath.Index(i).Child("target")
}
```